From 24ffb21ba81fcd1272aa3232c5f17cee6c7568ef Mon Sep 17 00:00:00 2001
From: Mark Reynolds <mreynolds@redhat.com>
Date: Thu, 4 Apr 2019 12:43:13 -0400
Subject: [PATCH] Ticket 50028 - ds-replcheck -y option throws usage error

Description:  Using the password file option (-y) causes
              a usage error to occur.  The arg validation
              needs to properly check for this option.

https://pagure.io/389-ds-base/issue/50028

Reviewed by: mreynolds(one line commit rule)
---
 ldap/admin/src/scripts/ds-replcheck | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ldap/admin/src/scripts/ds-replcheck b/ldap/admin/src/scripts/ds-replcheck
index f48240699..e18465dc0 100755
--- a/ldap/admin/src/scripts/ds-replcheck
+++ b/ldap/admin/src/scripts/ds-replcheck
@@ -1209,7 +1209,7 @@ def main():
     elif (args.mldif is None and
           (args.suffix is None or
            args.binddn is None or
-           (args.bindpw is None and args.prompt is False) or
+           (args.bindpw is None and (args.prompt is False and args.pass_file is None)) or
            args.murl is None or
            args.rurl is None)):
             print("\n-------> Missing required options for online mode!\n")
-- 
2.17.2

